src/test/java/com/amazon/opendistroforelasticsearch/knn/index/KNNESIT.java

```diff
@@ -35,6 +35,20 @@ public void testAddKNNDoc() throws Exception {
         addKnnDoc(INDEX_NAME, "1", FIELD_NAME, vector);
     }
 
+    /**
+     * Block adding new docs to KNN index when circuit breaker trips
+     */
+    public void testAddKNNDocBlockedWhenCbTrips() throws Exception {
+        createKnnIndex(INDEX_NAME, createKnnIndexMapping(FIELD_NAME, 2));
+        updateClusterSettings("knn.circuit_breaker.triggered", "true");
+
+        Float[] vector  = {6.0f, 6.0f};
+        ResponseException ex = expectThrows(
+                ResponseException.class, () -> addKnnDoc(INDEX_NAME, "1", FIELD_NAME, vector));
+        assertThat(EntityUtils.toString(ex.getResponse().getEntity()),
+                containsString("Cannot update KNN vector field when circuit breaker is triggered."));
+    }
```

Comment:
Can we unset ```knn.circuit_breaker.triggered``` flag and assert new doc could be added? Same with below test case.

Reply:
Sure.

Reply:
Np. I will add it. I think the test should pass as long as search does not go through the parser in vector mapper.

Reply:
thank you. Yes it should but this will act as a safety check for future code changes as well to ensure searches work fine even if CB triggered.